src/asmjit/core/virtmem.cpp

```diff
@@ -537,30 +554,30 @@ static Error detectShmStrategy(ShmStrategy* strategyOut) noexcept {
     return DebugUtils::errored(asmjitErrorFromErrno(e));
   }
   else {
-    munmap(ptr, vmInfo.pageSize);
+    munmap(ptr, pageSize);
     *strategyOut = ShmStrategy::kDevShm;
     return kErrorOk;
   }
 }
 #endif
 
-static Error getShmStrategy(ShmStrategy* strategyOut) noexcept {
+static Error getShmStrategy(ShmStrategy* strategyOut, MemoryFlags memoryFlags) noexcept {
```

Comment:
This is problematic - the SHM strategy is stored in a global, so if you call `getShmStrategy()` twice with different flags the detection only happens once and then reuses the detected value - this is to avoid calling mmap() every time a SHM strategy is requested.

This needs is to have two globals - keep/rename `globalShmStrategy` and add `hugePagesShmStrategy` for example - and decide which one to use based on the flags - once done this should not break anything

Reply:
Thank you for the context.  I will try out your suggestion of adding a separate function in the next revision I publish.